### PR TITLE
Use OpenSSL md5 if available

### PR DESCRIPTION
**Pull request by Learath2**
The change is originally by @heinrich5991, I also included the 2 helper functions but they are unused in teeworlds. I could remove those if you prefer.

### CODE REVIEW

File: src/engine/shared/network_token.cpp [hunk 1]
```diff
@@ -1,23 +1,17 @@
...
-	md5_state_t State;
-	unsigned int aDigest[4]; // make sure this is 16 byte
-
-	md5_init(&State);
-	md5_append(&State, (const md5_byte_t *)pData, Size);
-	md5_finish(&State, (md5_byte_t *)aDigest);
+	MD5_DIGEST Digest = md5(pData, Size);
 
-	return (aDigest[0] ^ aDigest[1] ^ aDigest[2] ^ aDigest[3]);
+	return (Digest.data[0] ^ Digest.data[1] ^ Digest.data[2] ^ Digest.data[3]);
```
Comment (oy) [inline src/engine/shared/network_token.cpp]: Should use uint* here, like before. `Digest.data` is uchar*.

Comment (Learath2) [reply src/engine/shared/network_token.cpp]: Oh, missed that you put the digest into an integer array.
I don't see a good way of doing that barring a `reinterpret_cast`. If you are okay with that I can commit it.
Or I can do something like `data[0] + data[1]<<0x8 + data[2] << 0x10 + data[3] << 0x18`

Comment (oy) [reply src/engine/shared/network_token.cpp]: For best readability it might be best to cast `Digest.data` to a uint* and then do the ^-operation with it.

**Comment by Learath2** [comment]
Not exactly the cleanest and safest of things to do, but I had to assume integers were 4 bytes.

**Comment by Dune-jr** [comment]
Couldn't you do a dynamic cast to int32 to make it safe? @Learath2

**Comment by heinrich5991** [comment]
`int` having 32 bit is a fair assumption that Teeworlds already makes in a lot of places.

Reinterpreting bytes is not the safest thing to do IMO; I find it cleaner to explicitly do the conversion (then it doesn't depend on endianness, has less chance of having UB, …).

**Comment by Dune-jr** [comment]
Thanks for the explanation @heinrich5991 :)

**Comment by oy** [comment]
Yeah, true @heinrich5991 an explicit conversation would be cleaner. Though teeworlds doesn't work on BE byte order anymore. Would be one less thing to fix :) 

@Learath2 you want to take care of it?
Maybe some helper function would be useful, there're some byte<>int conversions in the code.

**Comment by Learath2** [comment]
Sure I'll take are of it. Something like `int bytes_to_int(const unsigned char *data)`?

**Comment by heinrich5991** [comment]
Please include in the name whether you interpret the bytes in a big-endian or little-endian fashion. E.g. `bytes_be_to_int`.

**Comment by oy** [comment]
Yeah, sounds good 👍

**Pull request merged**